### PR TITLE
3d improvements / bugfixes

### PR DESCRIPTION
**Pull request by sashankaryal**
## Changelog

- Respect default up if it's present in plugin settings
- Persist 3D panel position and on/off state in browser storage
- Add fo3d json inspector
- Use modern [cameracontrols](https://github.com/yomotsu/camera-controls) instead of orbit controls. Also recommended by pmndrs/drei
- Fix gltf url parsing error
- Fix resource paths URL error
- Make grid more subtle with fading
- Increase default far clipping plane to 2500 from 2000
- Improve keyboard shortcut hints
- Set `THREE.DoubleSide` to true for all materials. While this addresses the face culling issue, we might want to revisit it later since it might have a performance impact for larger meshes, and so we might want to expose it as a material parameter.
- Store fo3d root in context
- Only show animation selector if more than one animation clip
- Add basic material controls (opacity and wireframe) for gltfs and fbxs that already have textures
- Turn off `scene.write()` default pretty-printing
- Simplify 3D action bar code, also fix transient bugs / QOL improvement
- Rename `Pointcloud` to PointCloud (**BREAKING CHANGE**)
- Upgrade three.js dependencies

## Summary by CodeRabbit

- **New Features**
  - Added an option to omit color controls in the `useMeshMaterialControls` function for enhanced customization based on material types.
  - Introduced `usePercolateMaterial` hook for setting opacity and wireframe properties across all materials in a 3D scene.
  - New function `useUrlModifier` for modifying URLs in the 3D fetching process.

- **Refactor**
  - Streamlined state management and removed unnecessary code in state handling for 3D scenes.
  - Simplified test cases for the `getFo3dRoot` function to improve root path extraction from file URLs.

- **Bug Fixes**
  - Corrected case sensitivity from "Pointcloud" to "PointCloud" in various utility functions, affecting scene summary and material handling.

- **Documentation**
  - Updated class references in the user guide to reflect renaming from `PointcloudMaterial` to `PointCloudMaterial`.

- **Style**
  - Standardized naming for `PointCloud` across the codebase to maintain consistency in 3D scene handling.

### CODE REVIEW

File: app/packages/looker-3d/src/fo3d/Gizmos.tsx [hunk 1]
```diff
@@ -70,50 +81,55 @@ export const Gizmos = () => {
...
+            side={DoubleSide}
+            cellSize={cellSize}
+            sectionSize={sectionSize}
+            sectionColor={GRID_SECTION_COLOR}
+            cellColor={GRID_CELL_COLOR}
+            fadeDistance={fadeDistance}
+            fadeStrength={1}
+            followCamera
+            cellThickness={0.5}
+            sectionThickness={0.8}
```
[the remark this answers is not in the document]
Comment (sashankaryal) [reply app/packages/looker-3d/src/fo3d/Gizmos.tsx]: This is an external component.

File: app/packages/looker-3d/src/action-bar/index.tsx [hunk 2]
```diff
@@ -1,19 +1,10 @@
...
-  ACTION_SET_EGO_VIEW,
-  ACTION_SET_PCDS,
-  ACTION_SET_POINT_SIZE,
-  ACTION_SET_TOP_VIEW,
-  ACTION_SHADE_BY,
-  ACTION_TOGGLE_BACKGROUND,
-  ACTION_VIEW_HELP,
-  ACTION_VIEW_JSON,
-} from "../constants";
+import { SET_EGO_VIEW_EVENT, SET_TOP_VIEW_EVENT } from "../constants";
```
[the remark this answers is not in the document]
Comment (sashankaryal) [reply app/packages/looker-3d/src/action-bar/index.tsx]: They are not imports from the same module, you're hallucinating.

File: docs/source/user_guide/using_datasets.rst [hunk 3]
```diff
@@ -4431,7 +4431,7 @@ to a FiftyOne 3D scene is shown below:
 
     import fiftyone as fo
 
-    fo_pcd = fo.Pointcloud(
+    fo_pcd = fo.PointCloud(
         "my-pcd", "/path/to/point-cloud.pcd", flag_for_projection=True
     )
```
[the remark this answers is not in the document]
Comment (sashankaryal) [reply docs/source/user_guide/using_datasets.rst:4437]: What....?

**Comment by swheaton** [comment]
Functional review:
1. Tested app config defaultUp, works, thanks!
2. shift-drag still doesn't work but is listed in the help so we should fix it or remove from help.
3. My scenes still start way too close, was this PR meant to fix that or no? (when camera position is null in fo3d)

> Only show animation selector if more than one animation clip

Also what does this mean, I thought we didn't support animation yet?

**Comment by swheaton** [comment]
> Functional review:
> 
> 1. Tested app config defaultUp, works, thanks!
> 2. shift-drag still doesn't work but is listed in the help so we should fix it or remove from help.
> 3. My scenes still start way too close, was this PR meant to fix that or no? (when camera position is null in fo3d)
>> Only show animation selector if more than one animation clip
> 
> Also what does this mean, I thought we didn't support animation yet?

#3 resolved now - maybe I didn't build app hard enough 🤷🏼 

A big problem is that PCD is not displaying in my dataset on this branch. I believe it was as of my review 3 days ago.

**Comment by sashankaryal** [comment]
> > Functional review:
> > 
> > 1. Tested app config defaultUp, works, thanks!
> > 2. shift-drag still doesn't work but is listed in the help so we should fix it or remove from help.
> > 3. My scenes still start way too close, was this PR meant to fix that or no? (when camera position is null in fo3d)
> > 
> > > Only show animation selector if more than one animation clip
> > 
> > 
> > Also what does this mean, I thought we didn't support animation yet?
> 
> bullet 3 resolved now - maybe I didn't build app hard enough 🤷🏼
> 
> A big problem is that PCD is not displaying in my dataset on this branch. I believe it was as of my review 3 days ago.

This PR introduced a breaking change; we're changing `Pointcloud` to `PointCloud` (and `PointcloudMaterial` to `PointCloudMaterial`), which is why. You'll want to either find+replace in your fo3ds, or generate them again.

**Comment by sashankaryal** [comment]
> Also what does this mean, I thought we didn't support animation yet?

We support animations in gltf and fbx, just not natively in fo3d.

**Review by ritch — APPROVED** [review]
Please add a comment describing the forked code approach so that we see it during merges as well as a jira issue to capture the plan to un-fork.

**Pull request merged**